apps/dotcms-ui/src/app/view/components/dot-theme-selector-dropdown/dot-theme-selector-dropdown.component.ts

```diff
@@ -182,24 +169,24 @@ export class DotThemeSelectorDropdownComponent
         }
     }
 
-    private getFilteredThemes(filter = '', offset = 0): void {
-        this.paginatorService.searchParam = filter;
+    private getFilteredThemes(offset = 0): void {
         this.setHostThemes(this.currentSiteIdentifier, this.currentOffset || offset);
     }
 
     private setHostThemes(hostId: string, offset: number = 0) {
+        this.siteService.getSiteById(hostId).subscribe((site: Site) => {
+            this.siteSelector.updateCurrentSite(site);
+        });
```

Comment:
I think we're subscribing multiples times here.

Reply:
tested and runs 1 time when dropdown is shown or site changed

Reply:
We're calling this method in the line 190, 147 and 133, with each call we subscribe, no?